packages/realm/src/Object.ts

```diff
@@ -66,7 +66,7 @@ export enum UpdateMode {
 }
 
 /** @internal */
-export type ObjCreator = () => [binding.Obj, boolean];
+export type ObjCreator = () => [binding.Obj, boolean, binding.TableRef?];
```

Comment:
Given that it's now 3 values, we could choose to return an object instead to clarify the return values.

Reply:
I agree, this is approaching (if its not already in) the territory that needs an object 👍

In this specific case, we might not need to pass the `TableRef` here at all, but we could instead pass a "rollback", "revert" or "delete" function, taking no arguments, which will call `table.removeObject(obj.key)`.

Reply:
Is the `TableRef` really optional?  Maybe i'm overlooking something, but the creator seems to always return a `TableRef`.